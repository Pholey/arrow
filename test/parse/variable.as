// Type and initializer are optional.
let value: int8 = 10;
let value = 2340;
let value;
