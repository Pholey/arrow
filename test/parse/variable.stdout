{
  "tag": "Module",
  "span": "test/parse/variable.as:1-4",
  "statements": [
    {
      "tag": "Variable",
      "span": "test/parse/variable.as:2:1-22",
      "name": "value",
      "mutable": false,
      "type": {
        "tag": "TypeName",
        "span": "test/parse/variable.as:2:12-16",
        "name": "int8"
      },
      "initializer": {
        "tag": "Integer",
        "span": "test/parse/variable.as:2:19-21",
        "value": "10"
      }
    },
    {
      "tag": "Variable",
      "span": "test/parse/variable.as:3:1-18",
      "name": "value",
      "mutable": false,
      "type": null,
      "initializer": {
        "tag": "Integer",
        "span": "test/parse/variable.as:3:13-17",
        "value": "2340"
      }
    },
    {
      "tag": "Variable",
      "span": "test/parse/variable.as:4:1-11",
      "name": "value",
      "mutable": false,
      "type": null,
      "initializer": null
    }
  ]
}
